src/logicblocks/event/processing/consumers/state.py

```diff
@@ -15,10 +18,22 @@
 from logicblocks.event.types.json import JsonValueConvertible
 
 
+def _support_bare_last_sequence_number_for_backwards_compat(
+    state: JsonObject, maybe_last_sequence_number: JsonValue | None
+) -> JsonObject:
+    """
+    Support bare last_sequence_number for backwards compatibility.
+    """
+    if isinstance(maybe_last_sequence_number, int):
+        state = dict(state)
+        state["last_sequence_number"] = maybe_last_sequence_number
+
+    return state
```

Comment:
Is this needed? Since we've maintained a converter for the previous approach, existing subscribers can use that and then future ones can use your new approach.

Reply:
The conversion is for the "state" property of the payload of the event, in the old method, the sequence number was at the top level, not nested under state

Reply:
I think the converter has kind of abstracted things so that the contents of consumer state is now a userland responsibility.

Reply:
Also, the converter produces a `JsonObject` not `EventConsumerState`, that `JsonObject` is added as the state value in `EventConsumerState`